src/NLog/Layouts/JsonLayout.cs

```diff
@@ -119,7 +125,17 @@ protected override string GetFormattedMessage(LogEventInfo logEvent)
             AppendIf(!this.SuppressSpaces, sb, " ");
             sb.Append("}");
 
-            return sb.ToString();
+
+            var result = sb.ToString();
+
+            if (!RenderEmptyLiteral)
+            {
+                if (result.Replace(" ", "") == "{}")
```

Comment:
This works, but it is a bit hacky ;)

I was thinking, postpone adding the `{` and `}` (eg. https://github.com/johnkors/NLog/blob/c2d5366d574c155b34591c96c31aa8c05c67730b/src/NLog/Layouts/JsonLayout.cs#L85) and add them if the object isn't empty.